pkg/distributor/distributor.go

```diff
@@ -340,129 +376,188 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	}
 	// Count the total samples in, prior to validation or deduplication, for comparison with other metrics.
 	incomingSamples.WithLabelValues(userID).Add(float64(numSamples))
+	// Count the total number of metadata in.
+	incomingMetadata.WithLabelValues(userID).Add(float64(len(req.Metadata)))
 
```

Comment:
I apologise for the diff here. I've had to switch validations around to ensure we cover the series/samples case first given metadata comes to a second plane. 

Although at the moment of writing a `WriteRequest` can only contain metadata or samples but not both, we believe that long-term that is not the case. We also don't want to couple the implementation here with what the remote-write systems do so we don't assume that. There's an exception on the ingesters but that is temporary until I'm done with the feature which should be sooner rather than later.

I have not changed _anything_ with regards to the functionality of the series and rather tried to build additional blocks for metadata.